test/test_torch.py

```diff
@@ -2386,39 +2386,259 @@ def _test_rpow(self, cast):
+        exps = [0, 1, 2, 4,
+                cast(torch.tensor((3, 3), dtype=torch.int8).random_(0, 10))]
+
+        for exp in exps:
+            _TestTorchMixin._check_against_np(self, tensor, exp)
 
     def test_int_pow(self):
         self._test_int_pow(self, lambda x: x)
+        self._test_short_pow(self, lambda x: x)
+        self._test_byte_pow(self, lambda x: x)
+        self._test_ubyte_pow(self, lambda x: x)
+
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    def test_int_tensor_pow_neg_ints(self):
+        ints = [torch.iinfo(torch.int32).min,
+                -3, -2, -1, 0, 1, 2, 3,
+                torch.iinfo(torch.int32).max]
+        neg_ints = [torch.iinfo(torch.int32).min, -3, -2, -1]
+
+        tensor = torch.tensor(ints, dtype=torch.int32)
+        nparr = np.array(ints, dtype=np.int32)
+        out = torch.empty_like(tensor)
+        for pow in neg_ints:
+            self.assertRaisesRegex(
+                ValueError,
+                "Integers to negative integer powers are not allowed.",
+                lambda: np.power(nparr, pow))
+            self.assertRaisesRegex(
+                RuntimeError,
+                "Integers to negative integer powers are not allowed.",
+                lambda: tensor.pow(pow))
+            self.assertRaisesRegex(
+                RuntimeError,
+                "Integers to negative integer powers are not allowed.",
+                lambda: tensor.pow_(pow))
+            self.assertRaisesRegex(
+                RuntimeError,
+                "Integers to negative integer powers are not allowed.",
+                lambda: torch.pow(tensor, pow))
+            self.assertRaisesRegex(
+                RuntimeError,
+                "Integers to negative integer powers are not allowed.",
+                lambda: torch.pow(tensor, pow, out=out))
+
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    def test_int_tensor_pow_non_neg_ints(self):
+        ints = [-3, -2, -1, 0, 1, 2, 3]
+        non_neg_ints = [0, 1, 2, 3]
+
+        tensor = torch.tensor(ints, dtype=torch.int32)
+        nparr = np.array(ints, dtype=np.int32)
+        out = torch.empty_like(tensor)
+        for pow in non_neg_ints:
+            expected = np.power(nparr, pow)
+
+            actual = tensor.pow(pow)
+            self.assertEqual(expected, actual.numpy())
+
+            actual = tensor.clone()
+            actual2 = actual.pow_(pow)
+            self.assertEqual(expected, actual.numpy())
+            self.assertEqual(expected, actual2.numpy())
+
+            actual = torch.pow(tensor, pow)
+            self.assertEqual(expected, actual.numpy())
+
+            actual2 = torch.pow(tensor, pow, out=actual)
+            self.assertEqual(expected, actual.numpy())
+            self.assertEqual(expected, actual2.numpy())
+
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    def test_extreme_int_tensor_pow_0_and_1(self):
```

Comment:
Please combine with `test_int_tensor_pow_non_neg_ints` using helper function which takes ints and pows.

Reply:
addressed